src/components/App/index.jsx

```diff
@@ -34,6 +34,8 @@ const App = () => {
     [questions, currQuestionIndex, setQuestions, setIsDisabled]
   );
 
+  // why my questions.length is becoming undefined after currQuestionIndex
+  // updates from 0 to 1 ?
```

Comment:
because your questions is initially an array that you get from data.math. 
However, in your checkAndUpdateAnswer, before you set new state, you are doing const questionsClone = { ...questions }; instead of const questionsClone = [ ...questions ]; so you're questions state becomes a regular object, which doesn't have a .length property.

Reply:
fixed it, thank you

Reply:
added comment above the checkAndUpdateAnswer